tests/devices/unit_tests/test_bimorph_mirror.py

```diff
@@ -29,117 +30,113 @@ def valid_bimorph_values(mirror: BimorphMirror) -> dict[int, float]:
 
 
 @pytest.fixture
-def mock_vtrgt_vout_propogation(mirror: BimorphMirror):
+def mirror_with_mocked_put(mirror: BimorphMirror):
+    async def busy_idle():
+        await asyncio.sleep(0)
+        set_mock_value(mirror.status, BimorphMirrorStatus.BUSY)
+        await asyncio.sleep(0)
+        set_mock_value(mirror.status, BimorphMirrorStatus.IDLE)
+
+    async def status(*_, **__):
+        asyncio.create_task(busy_idle())
+
+    for signal in walk_rw_signals(mirror).values():
+        callback_on_mock_put(signal, status)
+
     for channel in mirror.channels.values():
 
-        def effect(value: float, wait=False, signal=channel.output_voltage):
+        def vout_propogation_and_status(
+            value: float, wait=False, signal=channel.output_voltage
+        ):
```

Comment:
```suggestion
        def vout_propogation_and_status(value: float, wait: bool = False):
```

I don't think the third param was needed here?

Reply:
Resolved in person. Late binding closure issues fixed by definition an additional function within the loop.